contracts/access_control/lib.rs

```diff
@@ -34,6 +34,8 @@ mod access_control {
         Owner(AccountId),
         /// Indicates account can initialize a contract from a given code hash.
         Initializer(Hash),
+        /// Indicates account can add liquidity to a DEX contract (call certain functions)
+        LiquidityProvider(AccountId),
```

Comment:
Won't this force all users of this `Role` enum to handle this new variant as well? Even though they don't need it? Is there any versioning for this? If so, we should probably bump the version.

Reply:
it's not versioned, but how'd you version an enum? A separate crate just for enum's sake?

Reply:
I'm not sure whether such variant adding might break some compatibility, but maybe we will be more safe if we drop enum and use some plain string or byte slice as role identifier

Reply:
> How it's used so far does not force handling all variants - it's just a check for one of them, (for [example](https://github.com/Cardinal-Cryptography/aleph-node/blob/simple-dex/contracts/simple_dex/lib.rs?rgh-link-date=2022-09-13T08%3A00%3A45Z#L410))

That's how _you_ are using it _right now_. Maybe others would pattern match on the caller's `Role` and would have to handle all of the cases. Then, if they initially wrote a contract when `Role` had fewer entries, after you add a new one their code won't be compiling anymore.

Reply:
if we drop enum now they also won't be compiling anymore :rofl:

Reply:
Not if you version it and they use its semver properly :P